seacrowd/sea_datasets/id_sentiment_analysis/id_sentiment_analysis.py

```diff
@@ -0,0 +1,160 @@
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    seacrowd_schema_config: List[SEACrowdConfig] = []
+
+    for seacrowd_schema in _SUPPORTED_SCHEMA_STRINGS:
+
+        seacrowd_schema_config.append(
+            SEACrowdConfig(
+                name=f"{_DATASETNAME}_{seacrowd_schema}",
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} {seacrowd_schema} schema",
+                schema=f"{seacrowd_schema}",
+                subset_id=f"{_DATASETNAME}",
+            )
+        )
+
+    BUILDER_CONFIGS.extend(seacrowd_schema_config)
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "sentimen": datasets.Value("int32"),
+                    "tweet": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{str(TASK_TO_SCHEMA[Tasks.SENTIMENT_ANALYSIS]).lower()}":
+            features = schemas.text_features(label_names=[1, -1, 0])
+
+        else:
+            raise ValueError(f"Invalid config: {self.config.name}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        path = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "path": path,
+                },
+            ),
+        ]
+
+    def _generate_examples(self, path: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        idx = 0
+
+        if self.config.schema == "source":
+            df = pd.read_csv(path, delimiter="\t")
+
+            for _, row in df.iterrows():
+                yield idx, row.to_dict()
+                idx += 1
+
+        elif self.config.schema == f"seacrowd_{str(TASK_TO_SCHEMA[Tasks.SENTIMENT_ANALYSIS]).lower()}":
+            df = pd.read_csv(path, delimiter="\t")
+
+            df["id"] = df.index
+            df.rename(columns={"sentimen": "label"}, inplace=True)
+            df.rename(columns={"Tweet": "text"}, inplace=True)
```

Comment:
I am currently running into the following error as I run the test `python3 -m tests.test_seacrowd seacrowd/sea_datasets/id_sentiment_analysis/id_sentiment_analysis.py`

```
E
======================================================================
ERROR: runTest (__main__.TestDataLoader)
Run all tests that check:
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/builder.py", line 1687, in _prepare_split_single
    example = self.info.features.encode_example(record) if self.info.features is not None else record
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1866, in encode_example
    return encode_nested_example(self, example)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in encode_nested_example
    {
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in <dictcomp>
    {
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in zip_dict
    yield key, tuple(d[key] for d in dicts)
  File "/Users/yong/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in <genexpr>
    yield key, tuple(d[key] for d in dicts)
KeyError: 'tweet'
```

I think you need to lowercase the `Tweet` word here to fix the error.

Reply:
I forgot to rename the original label in the dataframe, fixed right away